File: Phase5-TDD/InvertedIndex.Test/QueryParserTest.cs
using System;
using System.Collections.Generic;
using Xunit;
using InvertedIndex.Search;
using InvertedIndex.Keywords;

namespace InvertedIndex.Test
{
    public class QueryParserTest : IDisposable
    {
        const string TEST_QUERY = "-exclude ordinary +union";
        IList<Keyword> _keywords;

        public QueryParserTest()
        {
            var parser = new QueryParser();
            _keywords = parser.ParseQuery(TEST_QUERY);
        }

        [Fact]
        public void ParseQueryTest1()
        {
            Assert.Equal("ordinary", _keywords[0].GetWord());
        }

        [Fact]
        public void ParseQueryTest2()
        {
            Assert.Equal("union", _keywords[1].GetWord());
        }

        [Fact]
        public void ParseQueryTest3()
        {
            Assert.Equal("exclude", _keywords[2].GetWord());
        }

        public void Dispose()
        {
            GC.Collect();
        }
    }
}

File: Inverted-Index/src/tests/QueryParserTest.java
package tests;

import org.junit.Test;

import keywords.Keyword;

import java.util.List;

import org.junit.After;
import org.junit.Assert;
import org.junit.Before;

import search.QueryParser;

public class QueryParserTest {
    
    private final String TEST_QUERY = "-exclude ordinary +union";
    private List<Keyword> keywords;

    @Before
    public void initKeywords() {
        var parser = new QueryParser();
        String query = TEST_QUERY;
        keywords = parser.parseQuery(query);
    }

    @Test
    public void parseQueryTest1() {

        Assert.assertEquals(keywords.get(0).getWord(), "ordinary");
    }

    @Test
    public void parseQueryTest2() {

        Assert.assertEquals(keywords.get(1).getWord(), "union");
    }

    @Test
    public void parseQueryTest3() {

        Assert.assertEquals(keywords.get(2).getWord(), "exclude");
    }

    @After
    public void dispose() {
        keywords = null;
    }
}